WordPress/Classes/Stores/RemoteFeatureFlagStore.swift

```diff
@@ -85,3 +92,43 @@ extension RemoteFeatureFlagStore {
         }
     }
 }
+
+fileprivate final class FetchRemoteFeatureFlagsOperation: AsyncOperation {
+    private let remote: FeatureFlagRemote
+    private let deviceID: String
+    private let completion: OperationCompletionHandler
+
+    typealias OperationCompletionHandler = (Result<[String: Bool], Error>) -> ()
+
+    enum OperationError: Error {
+        case operationCanceled
+    }
+
+    init(remote: FeatureFlagRemote, deviceID: String, completion: @escaping OperationCompletionHandler) {
+        self.remote = remote
+        self.deviceID = deviceID
+        self.completion = completion
+        super.init()
+    }
+
+    override func main() {
+        DDLogInfo("🚩 Updating Remote Feature Flags with Device ID: \(deviceID)")
+        remote.getRemoteFeatureFlags(forDeviceId: deviceID) { [weak self] result in
+            if self?.isCancelled == true {
+                self?.state = .isFinished
+                DDLogInfo("🚩 Feature flags update operation canceled")
+                self?.completion(.failure(OperationError.operationCanceled))
+                return
+            }
+            switch result {
+                case .success(let flags):
+                    DDLogInfo("🚩 Successfully updated local feature flags: \(flags.dictionaryValue)")
+                    self?.completion(.success(flags.dictionaryValue))
+                case .failure(let error):
+                    DDLogError("🚩 Unable to update Feature Flag Store: \(error.localizedDescription)")
+                    self?.completion(.failure(error))
+            }
+            self?.state = .isFinished
```

Comment:
Unless I'm reading the code wrong, if `self` is `nil`, nothing will happen other than console logs.

Have you considered returning early with `guard let self else { return }`?